benches/search_comparison.rs

```diff
@@ -167,111 +165,99 @@ fn search_bench_case<const MAX: usize, T, Coll>(
     setup_fun: impl Fn(Vec<T>) -> Coll,
     search_fun: impl Fn(&Coll, T) -> Option<&T>,
     group: &mut BenchmarkGroup<WallTime>,
-    size: &usize,
+    size: usize,
     duplicates: bool,
 ) where
-    T: TryFrom<usize> + Ord + Clone,
-    <T as TryFrom<usize>>::Error: core::fmt::Debug,
+    T: TryFrom<usize> + Ord + Copy,
 {
-    group.bench_with_input(BenchmarkId::new(name, size), size, |b, &size| {
+    group.bench_with_input(BenchmarkId::new(name, size), &size, |b, &size| {
         // increasing sequence of even numbers, bounded by MAX
         let iter = (0usize..)
             // Generate only even numbers to provide a ~50% hit ratio in the benchmark
             .map(|i| (i * 2) % MAX)
-            .map(|i| T::try_from(i).unwrap());
+            .map(|i| T::try_from(i).ok().unwrap());
 
         let v: Vec<T> = if duplicates {
-            iter
-                // Repeat each items 16 times
-                .flat_map(|i| std::iter::repeat(i).take(DUPLICATION_FACTOR))
+            iter.flat_map(|i| iter::repeat(i).take(DUPLICATION_FACTOR))
                 .take(size)
                 .collect()
         } else {
             iter.take(size).collect()
         };
 
-        let mut r = pseudorandom_iter::<T>(std::cmp::min(size * 2, MAX));
+        // to generate ~50% hit ratio generated number should be in the range 0..(size * 2) because
+        // test payload contains only even numbers: [0, 2, ..., 2 * size]
+        let mut r = pseudorandom_iter::<T>(MAX.min(size * 2));
         let c = setup_fun(v);
-        b.iter(|| {
-            let x = r.next().unwrap();
-            let _res = black_box(search_fun(&c, x));
-        })
+        b.iter(|| search_fun(&c, r.next().unwrap()))
```

Comment:
Do you think we ought to black box the second argument to seatch_fun here?

Reply:
What makes you think that? I don't see how optimizer can do any constant folding of `x`. Or am I missing something?

Reply:
I just never underestimate the power of future optimizers is all :p